gtsfm/utils/io.py

```diff
@@ -116,6 +118,42 @@ def read_bal(file_path: str) -> GtsfmData:
     return gtsfm_data
 
 
+def read_cameras_txt(fpath: str) -> Optional[List[Cal3Bundler]]:
+    """Read camera calibrations from a COLMAP-formatted cameras.txt file.
+
+    Reference: https://colmap.github.io/format.html#cameras-txt
+
+    Args:
+        fpaths: path to cameras.txt file
+
+    Returns:
+        calibrations: calibration object for each camera
+    """
+    if not Path(fpath).exists():
+        return None
+
+    with open(fpath, "r") as f:
+        lines = f.readlines()
+
+    num_cams = int(lines[2].replace('# Number of cameras: ', '').strip())
+    # should have one line per camera
+    assert len(lines[3:]) == num_cams
```

Comment:
maybe `assert len(lines)-3==num_cams`. Seems too inefficient to slice a list just to check its length.

Reply:
good suggestion -- updated